src/index.ts

```diff
@@ -1474,10 +1479,30 @@ export class GossipSub extends TypedEventEmitter<GossipsubEvents> implements Pub
     const now = Date.now()
     let doPX = this.opts.doPX
 
+    if (this.opts?.taggingEnabled ?? false) {
```

Comment:
I think this can ge done in a follow up PR if necessary considering not only is this an optional feature but it one immediate use case is where it can be used prune a connection via the connection manager in where that connection should not be gossiped to, in the case where it is enabled.